include/stdalgos/detail/merge.hpp

```diff
@@ -0,0 +1,180 @@
+                operator()(Scheduler&& sched, InputIt1 b1, InputIt2 e1,
+                    InputIt2 b2, InputIt1 e2, OutputIt d) const
+            {
+                // The sync_wait sender needs unwrapping since it returns an optional of tuple. And this
+                // overload returns an the output iterator.
+                return std::get<0>(stdexec::this_thread::sync_wait(
+                    merge_t{}(stdexec::transfer_just(
+                        std::forward<Scheduler>(sched), b1, e1, b2, e2, d)))
+                                       .value());
+            }
+
+            // Asynchronous overloads
+
+            // Use empty execution properties by default, forward to other overloads
+            template <stdexec::sender Sender>
+            stdexec::sender auto operator()(Sender&& sender) const
+            {
+                return merge_t{}(
+                    std::forward<Sender>(sender), execution_properties<>{});
+            }
+
+            // TODO: Add the overloads with tag_invocable and not tag_invocable with completion scheduler
+            template <typename InputIt1, typename InputIt2, typename OutputIt>
+            OutputIt merge_impl(InputIt1&& b1, InputIt1&& e1, InputIt2&& b2,
+                InputIt2&& e2, OutputIt&& d)
+            {
+                while (b1 != e1)
+                {
+                    if (*b2 < *b1)
+                    {
+                        d = *b2;    // equivalent: *d++ = *b2;
+                        ++b2;
+                    }
+                    else
+                    {
+                        d = *b1;    // equivalent: *d++ = *b1;
+                        ++b1;
+                    }
+                    if (b2 == e2)
+                        std::copy(b1, e1, d);
+                }
+                if (b1 == e1)
+                    std::copy(b2, e2, d);
+                return std::move(d);
+            }
+
+            // Default implementation
+            template <stdexec::sender Sender, typename... Properties>
+            requires
+                // clang-format off
+      (!stdexec::__tag_invocable_with_completion_scheduler<merge_t, stdexec::set_value_t, Sender>) &&
+      (!stdexec::tag_invocable<merge_t, Sender, execution_properties<Properties...> const&>)
+                // clang-format on
+                stdexec::sender auto
+                operator()(Sender&& sender,
+                    execution_properties<Properties...> const& exec_properties)
+            {
+                // TODO: Should a (completion) scheduler be required?
+                if constexpr (stdexec::__has_completion_scheduler<Sender,
+                                  stdexec::set_value_t>)
+                {
+                    stdexec::scheduler auto sched =
+                        stdexec::get_completion_scheduler<stdexec::set_value_t>(
+                            sender);
+                    return stdexec::let_value(std::forward<Sender>(sender),
```

Comment:
Please add a TODO here to change this back to a plain `then` once the `transfer` bug is fixed.